### DISCUSSION TITLE
Alternatives to deal with exceeding Serverless Functions size limit

### DISCUSSION THREAD

**Discussion by Leouss** [vercel/community#1982] [Help]
### Summary

We have been using ISR for our dynamic pages in NextJS (12.1.0) and we are recently getting this error when trying to deploy.
`NOW_SANDBOX_WORKER_MAX_LAMBDA_SIZE: The Serverless Function "_next/data/ywNlMclcwKa41MHuhq8Wy/projects/[id]/leads.json" is 50.27mb which exceeds the maximum size limit of 50mb.`
Leads.js is a page component in our project. This same error, with the same exceeding value, appeared for a different page component, and it was "solved" by changing to SSR, so it does not prerender on build, but apparently it was not enough. The prerender size will keep increasing since our users will likely add more data with time, but we would prefer to keep ISR or a similar fast option.

**Comment by amyegan** [comment]
Pagination is an option for long lists, which would allow you to return smaller chunks of data instead of sending the whole list at one time
